modules/bpi/bpi.module

```diff
@@ -397,6 +412,119 @@ function bpi_form_node_form_alter(&$form, &$form_state, $form_id) {
+ *   The syndicated images.
+ */
+function bpi_place_images(array &$form, $current_language, array $syndicated_images) {
+  $node = $form['#node'];
+  $image_field_names = bpi_fetch_image_fields($node->type);
+  if (!empty($image_field_names)) {
+    foreach ($syndicated_images as $image_type => $images) {
+      foreach ($image_field_names as $image_field_name) {
+        if (bpi_is_bpi_image_field($image_field_name, $node, $image_type)) {
+          $field = &$form[$image_field_name][$current_language];
+          $cardinality = isset($field['#cardinality']) ? $field['#cardinality'] : 1;
+          if ($cardinality < 0) {
+            $cardinality = PHP_INT_MAX;
+          }
+          $number_of_images_to_place = min(count($images), $cardinality);
+          for ($i = 0; $i < $number_of_images_to_place; $i++) {
+            if (is_object($images[$i]['@managed_file'])) {
+              $field[$i]['#default_value'] = array('fid' => $images[$i]['@managed_file']->fid);
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Replace inline images in content with references to syndicated images.
+ *
+ * @param string $field_name
+ *   The field name.
+ * @param string $content
+ *   The content.
+ * @param string $current_language
+ *    The current language.
+ * @param array $syndicated_images
+ *   The syndicated images.
+ *
+ * @return string
+ *   The content with inline images replaced by references to syndicated images.
+ */
+function bpi_place_inline_images($field_name, $content, $current_language, array $syndicated_images) {
+  if (!empty($syndicated_images)) {
+    $url_to_image_map = array();
+    foreach ($syndicated_images as $type => $images) {
+      foreach ($images as $image) {
+        $url_to_image_map[$image['path']] = $image;
+      }
+    }
+
+    // Find <img/> elements in content and replace with file placeholders.
+    $image_pattern = '@<img\s*[^>]*\ssrc=[\'"](?<url>[^\'">]+)[\'"][^>]*>@i';
+
+    $content = preg_replace_callback($image_pattern, function($matches) use ($current_language, $url_to_image_map) {
+      $url = $matches['url'];
+      if (isset($url_to_image_map[$url])) {
+        $image = $url_to_image_map[$url];
+        $file = $image['@managed_file'];
+
+        // Build file placeholder as used by the media_wysiwyg module.
+        $data = array(
+          'fid' => $file->fid,
+          'view_mode' => 'default',
+          'fields' => array(
+            'format' => 'default',
+            'field_file_image_alt_text[' . $current_language . '][0][value]' => isset($image['alt']) ? $image['alt'] : $image['name'],
+            'field_file_image_title_text[' . $current_language . '][0][value]' => isset($image['title']) ? $image['title'] : $image['name'],
+          ),
+          'type' => 'media',
+          'attributes' => array(
+            'width' => isset($image['width']) ? $image['width'] : 200,
+            'height' => isset($image['height']) ? $image['height'] : 200,
```

Comment:
Are `width` and `height` required attributes in the placeholder? If not I would think it better to leave them out instead of risking a distorted image. If the image on the other hand is very large we risk displaying it as is. Hmmm.

If you have had similar considerations then at least adding a comment why these defaults were chosen would be good.